sql/handler.cc

```diff
@@ -7326,6 +7326,24 @@ static bool check_period_field(const Create_field* f, const char* name,
     my_error(ER_WRONG_FIELD_SPEC, MYF(0), name);
     res= true;
   }
+  else if (f->vcol_info)
+  {
+    my_error(ER_PERIOD_FIELD_WRONG_ATTRIBUTES, MYF(0),
+             f->field_name.str, "VIRTUAL or GENERATED");
+    res= true;
+  }
+  else if (f->flags & EXPLICIT_NULL_FLAG)
```

Comment:
Because fields are `NULL` by default. 
As you preiously cited in  "Syntax bugs 1.3":
```
CD1 and CD2 both contain either an explicit or an implicit 
<column constraint definition> that
specifies NOT NULL NOT DEFERRABLE ENFORCED.
```
So it is allowed to be an implicit.
As far as i've got, timestamps in mariadb are also allowed to be implicit `NOT NULL`, but it has quite a hacky implementation, and there's something with sql modes.
In this case, I can't do the same hack, because the possibility of explicit `NOT NULL` existence is determined after the field is parsed.

So I need to know in logic, whether it was explicitly set or not

Reply:
Then you have 2 possibilities:
1. either disable implicit NULL, i.e. this query will fail:
```
create or replace table t (s date, e date, period for apptime(s,e));
```
with error "Column `s` can't be NULL".

2. or make period columns implicit NOT NULL, i.e.:
```
create or replace table t (s date, e date, period for apptime(s,e));
show create table t;
+-------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                                                                 |
+-------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| t     | CREATE TABLE `t` (
  `s` date DEFAULT NOT NULL,
  `e` date DEFAULT NOT NULL,
  PERIOD FOR `apptime` (`s`, `e`),
  CONSTRAINT `CONSTRAINT_1` CHECK (`s` < `e`)
) ENGINE=InnoDB DEFAULT CHARSET=latin1 |
+-------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

Current behavior of allowing period columns to be NULL is wrong.

Reply:
Indeed. Fixed.